api/src/main/java/org/eclipse/microprofile/concurrent/spi/ConcurrencyManagerBuilder.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.concurrent.spi;
+
+/**
+ * Use this class to configure instances of {@link ConcurrencyManager}.
+ */
+public interface ConcurrencyManagerBuilder {
+    
+    /**
+     * Use the specified {@link ThreadContextProvider} instances.
+     * @param providers the {@link ThreadContextProvider} instances to use.
+     * @return this builder
+     */
+    public ConcurrencyManagerBuilder withThreadContextProviders(ThreadContextProvider... providers);
+
+    /**
+     * Load all discoverable {@link ThreadContextProvider} instances via the {@link ServiceLoader}
+     * mechanism on the current thread-context {@link ClassLoader} (unless overridden by {@link #forClassLoader(ClassLoader)}).
+     * 
+     * @return this builder
+     */
+    public ConcurrencyManagerBuilder addDiscoveredThreadContextProviders();
```

Comment:
A minor comment on the builder - the presence of addDiscoveredThreadContextProviders without any parameters on a builder interface is a bit awkward.  Given this and the interaction between the two, we should consider if addDiscoveredThreadContextProviders and forClassLoader ought to be collapsed into a single builder method, such as,
```
withThreadContextProvidersFor(ClassLoader classLoader);
```

Reply:
I think it's meant to not repeat passing the `classLoader` for the rest of the providers, like when we add `withThreadContextPropagators()` and possibly more in the future.